motr/ut/idx_dix.c

```diff
@@ -1163,8 +1165,16 @@ static void dtm0_ut_send_redo(const struct m0_fid *ifid, uint32_t sdev_id,
 	m0_dix_fid_convert_dix2cctg(ifid, &cctg_fid, sdev_id);
 
 	dtm0_ut_cas_op_prepare(&cctg_fid, &cas_op, &cas_rec, key, val, &txr);
-
-	rc = m0_xcode_obj_enc_to_buf(&M0_XCODE_OBJ(m0_cas_op_xc, &cas_op),
+	dtm_payload.cdg_cas_op = cas_op;
+	dtm_payload.cdg_cas_opcode = opcode;
+	if (opcode == M0_CAS_DEL_FOP_OPCODE) {
```

Comment:
I think it's better to move this logic to dtm0_ut_cas_op_prepare() to have complete logic in a single place. Can check whether val is NULL instead of checking opcode in dtm0_ut_cas_op_prepare().